rfc/2025-07-22-conditional-enable.md

```diff
@@ -0,0 +1,195 @@
+
+### Errors when resource is not enabled
+
+There's some discussion about this [topic](https://github.com/opentofu/opentofu/issues/1306#issuecomment-2398120132), and the better compromise we found is that an enabled 
+resource can be `null` when it's disabled, but no errors are going to be shown statically,
+like when using `validate` or the language server, in an OpenTofu extension, like VSCode.
+Since we need to evaluate the `enabled` expression, these errors are going to be shown dynamically,
+when running `plan` or `apply`:
+
+```
+│ Error: Attempt to get attribute from null value
+│
+│   on main.tf line 19, in output "enabled_output":
+│   19:     value = null_resource.example.id
+│     ├────────────────
+│     │ null_resource.example is null
+│
+│ This value is null, so it does not have any attributes.
+```
+
+The only available data access patterns for possibly disabled resources are:
+
+- `null_resource.example != null ? null_resource.example.id : "default value"`
+- `try(null_resource.example.id, "default_value")`
+- `can(null_resource.example.id) ? null_resource.example.id : "default value"`
+
+Notice this can be extended by using custom provider functions. Let's say if someone would like to
+create a function that add a warning instead of raising an error when trying to access the attribute,
+they could do:
+
+- `provider::custom::warning_enabled(null_resource.example.id, "default value")
+
+For now, we will be offering the three options above. 
+`try` is the most concise option, but it should be used carefully since it can silently mask
+different errors not caused by null values access. `can` and `!= null` are more verbose but 
+are more reliable on doing what you're trying to do.
+
+Discarded options:
+
+- In a GitHub thread, someone mentioned using `?` to deal with disabled resources. Since this would be a HCL change, we discarded that option.
+
+### What happens when it's disabled
+
+Let's suppose we have created a resource using the `lifecycle -> enabled` field and then we want to disable it.
+The behavior is going to be the same as if a resource is being destroyed:
+
+```
+# aws_instance.demo_vm_2 will be destroyed
+# (because enabled is false)
+  - resource "aws_instance" "demo_vm_2" {
+      - ami                                  = "ami-07df274a488ca9195" -> null
+      - arn                                  = "arn:aws:ec2:eu-central-1:532199187081:instance/i-0b63433033e61d818" -> null
+      - associate_public_ip_address          = true -> null
+      - availability_zone                    = "eu-central-1b" -> null
+```
+
+### What types of variables are supported on the conditional
+
+There are a few types of values that cannot be used in this conditional:
+
+1. Unknown values, since they cannot be used until the apply phase
+2. Sensitive values
+3. Null values
+4. Conditionals that do not evaluate to true/bool values
+
+## Future considerations
+
```

Comment:
Please add some considerations around supporting count+enabled/for-each+enabled.